app/models/vm_or_template/operations/snapshot.rb

```diff
@@ -150,6 +150,18 @@ def remove_all_snapshots
     raw_remove_all_snapshots
   end
 
+  def remove_all_snapshots_queue(task_id = nil)
+    MiqQueue.put_unless_exists(
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => 'remove_all_snapshots',
+      :args        => [],
+      :role        => "ems_operations",
+      :zone        => my_zone,
+      :task_id     => task_id
```

Comment:
Instead of passing a task_id in I think you want to just do `MiqTask.generic_action_with_callback` which will create the task and when the miq_queue method finishes it updates the task.

Reply:
Ah I see you're following the pattern of the rest of the `remove_snapshot_queue` methods, I'm not sure that the task will be updated when the queue item finishes

Reply:
So I don't think this task_id is what you're thinking it is, https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_action.rb#L635-L640